### PR TITLE
Selhosted version Improvements  and additional features

### PR DESCRIPTION
**Pull request by tckb**
This PR has the following changes

- [x] Formatting

**Selfhosting setup change/updates**

- [x] Clickhouse migration updates
- [x] update readme to add basic guides for installing/setup for postgres/clickhouse (closes https://github.com/plausible/analytics/issues/84) 
- [x] More UI tweaks for selhosting setup
     - [x]  added config for disabling subscription
     - [x]  hide  "trial" days banner
     - [x]  hide  subscription plans (closes https://github.com/plausible/analytics/issues/87)

**Bug fix /New Features**

- [x] Show Direct Traffic in dashboard (closes https://github.com/plausible/analytics/issues/187)
- [x] PR comments

### CODE REVIEW

File: lib/plausible_web/controllers/api/external_controller.ex [hunk 1]
```diff
@@ -135,6 +135,7 @@ defmodule PlausibleWeb.Api.ExternalController do
 
   defp parse_body(conn) do
     {:ok, body, _conn} = Plug.Conn.read_body(conn)
+    IO.inspect(body)
```
Comment (ukutaht) [inline lib/plausible_web/controllers/api/external_controller.ex:138]: Probably meant to remove this?

File: lib/plausible/stats/clickhouse.ex [hunk 2]
```diff
@@ -276,17 +278,53 @@ defmodule Plausible.Stats.Clickhouse do
...
+               fragment("uniq(user_id) as count")},
+            where: e.referrer_source == ""
+        )
+      if no_referrers|> hd |> Map.get("count") > 0, do: referrers ++ no_referrers, else: []
+    else
+      referrers
+    end
+  end
+
+  defp bounce_rates_by_referrer_source(site, query) do
```
Comment (ukutaht) [inline lib/plausible/stats/clickhouse.ex:311]: I don't think this is used anywhere

File: lib/plausible_web/controllers/api/external_controller.ex (hunk 1 above)
Comment (tckb) [reply lib/plausible_web/controllers/api/external_controller.ex:138]: Damn! yes, will do

File: lib/plausible/stats/clickhouse.ex (hunk 2 above)
Comment (tckb) [reply lib/plausible/stats/clickhouse.ex:311]: will get the latest changes, there seems to be some updates from you.

**Comment by tckb** [comment]
PR Comments addressed @ukutaht

**Pull request merged**